### ISSUE TITLE
Users unable to reset password

### ISSUE THREAD

**Issue by helen-d-little** [alan-turing-institute/data-safe-haven#1676]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** macOS
- **Data Safe Haven version:** 4.1.0

## :cactus: Powershell module versions

```
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.Monitor module version: 4.2.0
2023-11-29 13:06:43 [SUCCESS]: [✔] Powershell-Yaml module version: 0.4.2
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.RecoveryServices module version: 5.4.1
2023-11-29 13:06:43 [SUCCESS]: [✔] Microsoft.Graph.Authentication module version: 1.21.0
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.PrivateDns module version: 1.0.3
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.Dns module version: 1.1.2
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.KeyVault module version: 4.9.1
2023-11-29 13:06:43 [SUCCESS]: [✔] Microsoft.Graph.Applications module version: 1.21.0
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.Network module version: 5.3.0
2023-11-29 13:06:43 [SUCCESS]: [✔] Az.Automation module version: 1.9.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Microsoft.Graph.Users module version: 1.21.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.Resources module version: 6.5.1
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.Accounts module version: 2.11.1
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.MonitoringSolutions module version: 0.1.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.OperationalInsights module version: 3.1.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.DataProtection module version: 0.4.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.Storage module version: 4.7.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Microsoft.Graph.Identity.DirectoryManagement module version: 1.21.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Az.Compute module version: 5.3.0
2023-11-29 13:06:44 [SUCCESS]: [✔] Poshstache module version: 0.1.10
```

## :no_entry_sign: Describe the problem

From approximately 20th November 2023 users (both existing and new) are unable to reset their password. When they try they get the following error message.
![error-message_image](https://github.com/alan-turing-institute/data-safe-haven/assets/46891265/c6149a2c-8528-47e4-ad49-028866978fd9)

A fix was attempted following the documentation ([linked here](https://data-safe-haven.readthedocs.io/en/latest/roles/system_manager/manage_users.html#password-reset-failure)) but was unsuccessful.

Viewing the firewall settings via the Azure portal (`Resource Groups > RG_SHM_<SHM_ID>_NETWORKING > FIREWALL-SHM-<SHMID> | Rules (classic)` ) The `AllowExternalAzureADConnectPasswordReset` Target FQDNs did not match those specified in the DSH - 4.1.0 - repo `deployment/safe_haven_management_environment/network_rules/shm-firewall-rules.json`. 

To update the firewall requirements of the active SHM to those specified in the 4.1.0 release;

`./Setup_SHM_Firewall.ps1` was run from the 

`data-safe-haven/deployment/safe_haven_management_environment/setup` directory.

Changes took about 5 minutes to take effect, users have reported they can update their passwords.

Note this was addressed in issue #1467 and the fix was expected to have propagated through to this release.

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
Your log details here
```

</details>

**Comment by jemrobinson** [comment]
@craddm : Should this have been fixed by your recent firewall rule changes?

**Comment by craddm** [comment]
This was fixed a while back, yes.

**Comment by jemrobinson** [comment]
Fix should be present in 4.2.0.
